File: kotlinExamples/src/test/java/com/example/MoneyTests.java
package com.example;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

import java.math.BigDecimal;
import java.util.Currency;
import java.util.Locale;
import org.junit.jupiter.api.Test;

public class MoneyTests {

  @Test
  public void add_money() {
    Money money1 = Money.of(BigDecimal.valueOf(20.24), Currency.getInstance(Locale.US));
    Money money2 = Money.of(BigDecimal.valueOf(29.76), Currency.getInstance(Locale.US));
    Money expected = Money.of(BigDecimal.valueOf(50), Currency.getInstance(Locale.US));
    assertEquals(money1.add(money2), expected);
  }

  @Test
  public void add_money_fail() {
    Money money1 = Money.of(BigDecimal.valueOf(20.24), Currency.getInstance(Locale.CANADA));
    Money money2 = Money.of(BigDecimal.valueOf(29.76), Currency.getInstance(Locale.CHINA));
    assertThrows(
        IllegalArgumentException.class,
        () -> money1.add(money2)
    );
  }

}

File: kotlinExamples/src/test/kotlin/gradle/project/kotlinExamples/MoneyTests.kt
package gradle.project.kotlinExamples

import java.math.BigDecimal
import java.util.*
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFailsWith

class MoneyTests {

    @Test
    fun money_add() {
        val money1: Money = Money.of(BigDecimal(20), Currency.getInstance(Locale.FRANCE))
        val money2: Money = Money.of(BigDecimal(30), Currency.getInstance(Locale.FRANCE))
        val expected: Money = Money.of(BigDecimal(50), Currency.getInstance(Locale.FRANCE))
        assertEquals(
            money1.add(money2),
            expected
        )
    }

    @Test
    fun money_add_failure() {
        val money1: Money = Money.of(BigDecimal(20), Currency.getInstance(Locale.US))
        val money2: Money = Money.of(BigDecimal(30), Currency.getInstance(Locale.FRANCE))
        assertFailsWith<IllegalArgumentException> {
            money1.add(money2)
        }
    }

}